lib/ansible/modules/cloud/amazon/cloudwatchlogs_log_group.py

```diff
@@ -0,0 +1,298 @@
+    kms_key_id: arn:aws:kms:region:account-id:key/key-id
+
+- cloudwatchlogs_log_group:
+    state: absent
+    log_group_name: test-log-group
+
+'''
+
+RETURN = '''
+log_groups:
+    description: Return the list of complex objetcs representing log groups
+    returned: success
+    type: complex
+    contains:
+        log_group_name:
+            description: The name of the log group.
+            returned: always
+            type: string
+        creation_time:
+            description: The creation time of the log group.
+            returned: always
+            type: integer
+        retention_in_days:
+            description: The number of days to retain the log events in the specified log group.
+            returned: always
+            type: integer
+        metric_filter_count:
+            description: The number of metric filters.
+            returned: always
+            type: integer
+        arn:
+            description: The Amazon Resource Name (ARN) of the log group.
+            returned: always
+            type: string
+        stored_bytes:
+            description: The number of bytes stored.
+            returned: always
+            type: string
+        kms_key_id:
+            description: The Amazon Resource Name (ARN) of the CMK to use when encrypting log data.
+            returned: always
+            type: string
+'''
+
+try:
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+import traceback
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import _camel_to_snake, camel_dict_to_snake_dict, boto3_conn, ec2_argument_spec, get_aws_connection_info
+
+try:
+    import botocore
+except ImportError:
+    pass  # will be detected by imported HAS_BOTO3
+
+
+def create_log_group(client, log_group_name, kms_key_id, tags, retention, module):
+    try:
+        request = {
+            'logGroupName': log_group_name
+        }
+        if kms_key_id:
+            request['kmsKeyId'] = kms_key_id
+        if tags:
+            request['tags'] = tags
+
+        client.create_log_group(**request)
+
+        if retention:
+            input_retention_policy(client=client,
+                                   log_group_name=log_group_name,
+                                   retention=retention, module=module)
+
+        desc_log_group = describe_log_group(client=client,
+                                            log_group_name=log_group_name,
+                                            module=module)
+
+        if 'logGroups' in desc_log_group:
+            for i in desc_log_group['logGroups']:
+                if log_group_name == i['logGroupName']:
+                    return i
+                else:
+                    module.fail_json(msg="The aws CloudWatchLogs log group was not created. \n please try again!")
+
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg=e.response, exception=traceback.format_exc(),
```

Comment:
It's preferable to have msg to be something specific short and specific about what call failed - such as "Unable to create log group: {0}".format(log_group_name) since e.response is being used twice. Same for any following too.